flask_sqlalchemy/__init__.py

```diff
@@ -372,6 +373,47 @@ def next_num(self):
             return None
         return self.page + 1
 
+    @staticmethod
+    def _serialize_item(value):
+        """Handles specific JSON serialization of different datatypes"""
+        if isinstance(value, bool):
+            if value:
+                return 'true'
+            return 'false'
+        if isinstance(value, int) or isinstance(value, float):
+            return value
+        if isinstance(value, str):
+            return value
+        if isinstance(value, dt.datetime):
+            return value.isoformat()
+        if isinstance(value, list):
+            return [Pagination._serialize_item(i) for i in value]
+        if isinstance(value, dict):
+            return {Pagination._serialize_item(k): Pagination._serialize_item(v)
+                    for k, v in value.items()}
```

Comment:
nit: use the `iteritems` from _compat.py to avoid wasting resources in Python 2)